Sources/PackageGraph/PackageGraph+Loading.swift

```diff
@@ -652,3 +653,23 @@ fileprivate func findCycle(
     // Couldn't find any cycle in the graph.
     return nil
 }
+
+
+// TODO: model package location better / encapsulate into a new type (PackageLocation) so that such comparison is reusable
+// additionally move and rename CanonicalPackageIdentity to become a detail function of the PackageLocation abstraction, as it is not used otherwise
+struct LocationComparator {
```

Comment:
Would this be better as free function, or is the idea to extend this struct in the future along the lines of the TODO?

Reply:
yes, this will be extended / moved in follow up PR

Reply:
Great, that's what I thought.